packages/front-end/services/experiments.ts

```diff
@@ -245,6 +245,25 @@ export function applyMetricOverrides(
   return { newMetric, overrideFields };
 }
 
+export function isExpectedDirection(
+  stats: SnapshotMetric,
+  metric: MetricInterface
+): boolean {
+  const expected: number = stats?.expected ?? 0;
+  return metric.inverse ? expected < 0 : expected > 0;
+}
+
+export function isStatSig(
+  stats: SnapshotMetric,
+  pValueThreshold: number
+): boolean {
+  const pValue: number = stats?.pValue ?? 1;
+  return pValue < pValueThreshold;
+}
+
 export function pValueFormatter(pValue: number) {
+  if (pValue === undefined) {
+    return "P-value missing";
+  }
   return pValue < 0.001 ? "<0.001" : pValue.toFixed(3);
```

Comment:
This may be unrelated to your current changes, but this 0.001 value looks like a magic number. Is it possible this could change and we'd have to find and replace all instances of 0.001?

Reply:
This isn't as important as the pValueThreshold, for example, and just helps simplify the output rather than returning scientific notation or something truncated to 0.000 or something similarly unclear. This could be better handled in the front end as some general significant figures/rounding control, but for now I'm not too concerned about this.